ext/standard/array.c

```diff
@@ -6389,3 +6389,155 @@ PHP_FUNCTION(array_combine)
 	} ZEND_HASH_FOREACH_END();
 }
 /* }}} */
+
+static inline void php_array_until(zval *return_value, HashTable *htbl, zend_fcall_info fci, zend_fcall_info_cache fci_cache, int stop_value) /* {{{ */
```

Comment:
In `.c` files the `inline` should either be removed, or should be changed into `zend_always_inline` depending on intent.

```suggestion
static void php_array_until(zval *return_value, HashTable *htbl, zend_fcall_info fci, zend_fcall_info_cache fci_cache, int stop_value) /* {{{ */
```

Reply:
I was basing this on the surrounding code, such as `php_search_array` used for in_array and array_search

I'd assume there would be a place where you'd want "inline depending on compiler's best guess at optimization level" but I'm not familiar with the why of zend_always_inline vs inline

```
/* void php_search_array(INTERNAL_FUNCTION_PARAMETERS, int behavior)
 * 0 = return boolean
 * 1 = return key
 */
static inline void php_search_array(INTERNAL_FUNCTION_PARAMETERS, int behavior) /* {{{ */
{
```

Reply:
No need to specify `inline`; a `static` function in a `.c` file is enough. If you want to be sure it's inlined, that's when you throw on `zend_always_inline`.